packages/lwc-language-server/src/component-indexer.ts

```diff
@@ -122,6 +122,43 @@ export default class ComponentIndexer extends BaseIndexer {
         fsExtra.writeFileSync(indexPath, indexJsonString);
     }
 
+    // This is a temporary solution to enable automated LWC module resolution for TypeScript modules.
+    // It is intended to update the path mapping in the .sfdx/tsconfig.sfdx.json file.
+    // TODO: Once the LWC custom module resolution plugin has been developed in the language server
+    // this can be removed.
+    updateSfdxTsConfigPath(): void {
+        const sfdxTsConfigPath = normalize(`${this.workspaceRoot}/.sfdx/tsconfig.sfdx.json`);
+        if (fs.existsSync(sfdxTsConfigPath)) {
+            try {
+                const sfdxTsConfig = utils.readJsonSync(sfdxTsConfigPath);
+                // The assumption here is that sfdxTsConfig will not be modified by the user as
+                // it is located in the .sfdx directory.
+                sfdxTsConfig.compilerOptions.paths['c/*'] = this.tsConfigPathMappingFiles;
+                utils.writeJsonSync(sfdxTsConfigPath, sfdxTsConfig);
+            } catch (err) {
+                console.error(err);
+            }
+        }
+    }
+
+    get tsConfigPathMappingFiles(): string[] {
+        let files: string[] = [];
+        if (this.workspaceType === WorkspaceType.SFDX) {
+            const sfdxSource = normalize(`${this.workspaceRoot}/${this.sfdxPackageDirsPattern}/**/*/lwc/*/*.{js,ts}`);
+            const filePaths = sync(sfdxSource, { stats: true });
```

Comment:
This does a fast glob search through the file system with the given glob pattern.

Note that we're not reading the actual files, just scanning the file system.

This will be done each time the mapping needs to be updated.

Performance should be fine in this scenario, I believe even with a few thousand components the search should only take a new miliseconds.